packages/adblocker-electron-preload/src/index.ts

```diff
@@ -7,24 +7,28 @@
  */
 
 import { ipcRenderer } from 'electron';
+import { DOMMonitor, IBackgroundCallback } from '@cliqz/adblocker-content';
 
-import {
-  DOMMonitor,
-  IBackgroundCallback,
-  IMessageFromBackground,
-  injectScript,
-} from '@cliqz/adblocker-content';
+type CosmeticFiltersResponse = {
+  active: boolean;
+  error?: string;
+};
 
-function getCosmeticsFiltersFirst(): string[] | null {
-  return ipcRenderer.sendSync('get-cosmetic-filters-first', window.location.href);
-}
-function getCosmeticsFiltersUpdate(data: Omit<IBackgroundCallback, 'lifecycle'>) {
-  ipcRenderer.send('get-cosmetic-filters', window.location.href, data);
+function injectCosmeticFilters(
+  data?: Omit<IBackgroundCallback, 'lifecycle'>,
+): Promise<CosmeticFiltersResponse> {
+  return ipcRenderer.invoke(
+    '@ghostery/adblocker/inject-cosmetic-filters',
+    window.location.href,
+    data,
+  );
 }
 
 if (window === window.top && window.location.href.startsWith('devtools://') === false) {
-  (() => {
-    const enableMutationObserver = ipcRenderer.sendSync('is-mutation-observer-enabled');
+  (async () => {
```

Comment:
Not sure if there is a strong reason to make this async. 
For the `injectCosmeticFilters` you can use regular `then`.

Reply:
I did a general API overhaul to switch to using `invoke` in general. This is a much more modern and overall better approach. Please correct me if I'm wrong.